libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/FreeDriveLocationUpdater.kt

```diff
@@ -52,18 +61,28 @@ internal class FreeDriveLocationUpdater(
                 }
             }, 1500, 1000, TimeUnit.MILLISECONDS)
         }
+
+        if (electronicHorizonFuture == null) {
+            electronicHorizonFuture = executorService.scheduleAtFixedRate({
```

Comment:
We're running this in a `newSingleThreadScheduledExecutor` https://github.com/mapbox/mapbox-navigation-android/blob/825a8cc4bc6519bc31a07088eda3599b290b9ac4/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigation.java#L1043 could this be a problem? Wondering if could interfere with https://github.com/mapbox/mapbox-navigation-android/blob/825a8cc4bc6519bc31a07088eda3599b290b9ac4/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/FreeDriveLocationUpdater.kt#L43 task.

Reply:
I think they will be queued in a worst case

Reply:
Yeah, that's what I was thinking especially in the context of queueing https://github.com/mapbox/mapbox-navigation-android/blob/f46c18d4ab60b44417fd8090c62b25600435c82b/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/FreeDriveLocationUpdater.kt#L46-L51 We don't want location fixes to be queued, right? In other words, location fixes returned shouldn't be impacted by the `getElectronicHorizon` calls. What do you think?

Reply:
let's use `Executors.newScheduledThreadPool(2)` here?

Reply:
or 2 separate `Executors.newSingleThreadScheduledExecutor()` ?

Reply:
discussed, use 1 executor with 2 threads